### PR TITLE
sys-cluster/hpx: Bump version

### PR DESCRIPTION
**Pull request by shifty91**
Add the current HPX version: http://stellar-group.org/2020/02/hpx-1-4-1-released/. 
The documentation generation requires an update to date dev-python/breathe version.

Package-Manager: Portage-2.3.89, Repoman-2.3.20
Signed-off-by: Kurt Kanzenbach <kurt@kmk-computers.de>

### CODE REVIEW

File: dev-python/breathe/breathe-4.14.1.ebuild [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+EAPI=7
+
+PYTHON_COMPAT=( python3_{6..8} )
+DISTUTILS_USE_SETUPTOOLS=rdepend
+
+inherit distutils-r1
+
+DESCRIPTION="Sphinx Doxygen renderer"
+HOMEPAGE="https://breathe.readthedocs.io/en/latest/"
+SRC_URI="mirror://pypi/${PN:0:1}/${PN}/${P}.tar.gz"
```
Comment (juippis) [inline dev-python/breathe/breathe-4.14.1.ebuild]: https://github.com/michaeljones/breathe looks like at least Github has tests listed, although I didnt download a tarball to confirm they are packaged. Could you check?

File: sys-cluster/hpx/hpx-1.4.1.ebuild [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+			dev-python/sphinx[${PYTHON_MULTI_USEDEP}]
+			dev-python/sphinx_rtd_theme[${PYTHON_MULTI_USEDEP}]
+			>=dev-python/breathe-4.14[${PYTHON_MULTI_USEDEP}]
+		')
+	)
+	test? ( ${PYTHON_DEPS} )
+"
+RDEPEND="
+	${PYTHON_DEPS}
+	dev-libs/boost
```
Comment (juippis) [inline sys-cluster/hpx/hpx-1.4.1.ebuild]: Are you sure this isn't linked? Boost is usually quite ABI sensitive, please check.

File: sys-cluster/hpx/hpx-1.4.1.ebuild [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+	sys-apps/hwloc
+	sys-libs/libunwind
```
Comment (juippis) [inline sys-cluster/hpx/hpx-1.4.1.ebuild]: Is this really an unconditional dependency on every architechture? Seems weird.

File: dev-python/breathe/breathe-4.14.1.ebuild (hunk 1 above)
Comment (shifty91) [reply dev-python/breathe/breathe-4.14.1.ebuild]: The tests are not packaged.

File: sys-cluster/hpx/hpx-1.4.1.ebuild (hunk 2 above)
Comment (shifty91) [reply sys-cluster/hpx/hpx-1.4.1.ebuild]: Boost is linked e.g. into the libhpx.so. The original ebuild used `dev-libs/boost:=`. Is that the correct way?

File: sys-cluster/hpx/hpx-1.4.1.ebuild (hunk 3 above)
Comment (shifty91) [reply sys-cluster/hpx/hpx-1.4.1.ebuild]: No, it's not. The documentation for 1.4.1 states that libunwind is needed whenever google-perftools is used on x86_64 (https://stellar-group.github.io/hpx-docs/latest/html/manual/building_hpx.html). The ebuild google-perftools already includes the dependency to libunwind. So, that one here can be removed.

File: sys-cluster/hpx/hpx-1.4.1.ebuild (hunk 2 above)
Comment (shifty91) [reply sys-cluster/hpx/hpx-1.4.1.ebuild]: Hm, now I get it (https://wiki.gentoo.org/wiki/Sub-slots_and_Slot-Operators). Should be fixed now. Thanks!

**Comment by shifty91** [comment]
Why does it fail? Out of memory, maybe? hpx is c++ heavy. It works for me with twice as much RAM in G as cores (16 cores -> 32 GiB RAM). But, what can I do about it?

**Comment by juippis** [comment]
Oh, wow, it does indeed. Never thought I'd see that happen.

You may want to utilize *check-reqs.eclass* because it's better to stop the user at `pkg_setup` instead of leaving them with filled tmpfs for example. I don't have a straight example how to use it with scaling values, but I imagine you could do it in steps. Something like 
```
if [[ ${MAKEOPTS} -ge j4 ]]; then
	CHECKREQS_MEMORY=8G

elif [[ ${MAKEOPTS} -ge j8 ]]; then
	CHECKREQS_MEMORY=16G

elif [[ ${MAKEOPTS} -ge j12 ]]; then
	CHECKREQS_MEMORY=24G

elif [[ ${MAKEOPTS} -ge j16 ]]; then
	CHECKREQS_MEMORY=32G

else
	CHECKREQS_MEMORY=2G
```

probably there are better ways, but you get the idea I'm after. Maybe you can even just take the number from MAKEOPTS and multiply it by 2 and add that to `CHECKREQS_MEMORY`.

**Comment by shifty91** [comment]
1G RAM seems to be enough per job. Added that to the ebuild. Also switched to `cmake_src_test` and called python setup unconditionally, which fixes builds without `doc` and `test`.
But, I have no idea why the warning about the boost root is issued. Boost root is actually considered and this only happens when the tests are executed...

**Comment by shifty91** [comment]
@juippis Found the boost cmake problem. Should be fixed now. Thanks for the review.

**Comment by shifty91** [comment]
The tests shouldn't be executed in parallel. Anyway, the ebuild should take care of that. Yes, that needs further research. Thanks!

**Pull request closed without merging**